client/client.go

```diff
@@ -363,6 +363,7 @@ func (c *client) tsLoop() {
 			if err != nil {
 				select {
 				case <-loopCtx.Done():
+					cancel()
```

Comment:
why not calling cancel in defer?

Reply:
There will be a lint `this return statement may be reached without using the cancel var defined`.